hypervisor/common/hypercall.c

```diff
@@ -400,10 +447,14 @@ int32_t hcall_inject_msi(struct acrn_vm *vm, uint16_t vmid, uint64_t param)
 			pr_err("%s: Unable copy param to vm\n", __func__);
 			ret = -1;
 		} else {
-			ret = vlapic_intr_msi(target_vm, msi.msi_addr, msi.msi_data);
+			/* For target cpu with lapic pt, send ipi instead of injection via vlapic */
+			if (is_lapic_pt(target_vm)) {
+				inject_msi_lapic_pt(target_vm, &msi);
+				ret = 0;
+			} else {
+				ret = vlapic_intr_msi(target_vm, msi.msi_addr, msi.msi_data);
```

Comment:
here should we still call vlapic_intr_msi and we can distinguish lapic_pt in the vlapic_intr_msi?

Reply:
Thanks. Merged.